src/Stratis.Bitcoin.Features.SmartContracts/MempoolRules/CanGetSenderMempoolRule.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+using Stratis.SmartContracts.Core;
+using Stratis.SmartContracts.Core.Util;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.MempoolRules
+{
+    public class CanGetSenderMempoolRule : MempoolRule
+    {
+        private readonly ISenderRetriever senderRetriever;
+
+        // TODO: It does not appear this was getting called in the smart contract mempool validator. Should it be?
```

Comment:
Actually, the corresponding consensus rule is full validation, and therefore should be getting injected as part of the `ContractTransactionFullValidationMempoolRule` rule checks.